Dockerfile

```diff
@@ -1,10 +1,44 @@
-FROM elixir:1.10-alpine
-RUN mix local.hex --force
-RUN mix local.rebar --force
+# ---- Build Stage ----
+FROM elixir:1.10-alpine AS builder
+
+# Set environment variables for building the application
+ENV MIX_ENV=prod \
+    TEST=1 \
+    LANG=C.UTF-8
+
+# Install hex and rebar
+RUN mix local.hex --force && \
+    mix local.rebar --force
+
+
+# Create the application build directory
+RUN mkdir /app
 WORKDIR /app
-ADD . /app
+
+# Copy over all the necessary application files and directories
+COPY config ./config
+COPY lib ./lib
+COPY priv ./priv
+COPY mix.exs .
+COPY mix.lock .
+
+# Fetch the application dependencies and build the application
 RUN mix deps.get
 RUN mix deps.compile
-RUN mix compile
-CMD mix phx.server
-EXPOSE 4000
+RUN mix release
+
+# ---- Application Stage ----
+FROM alpine AS runtime
+
+ENV LANG=C.UTF-8
+
+# Install openssl
+RUN apk update && apk add openssl ncurses-libs
+
+# Copy over the build artifact from the previous step
+COPY --from=builder /app/_build .
+COPY entrypoint.sh .
+
+RUN ls -la
+
+ENTRYPOINT ["sh", "./entrypoint.sh"]
```

Comment:
no need to call sh

Reply:
actually yes, it's needed for the entrypoint command, otherwise it doesn't work

Reply:
the first argument needs to be executable, as explained [in the docs](https://docs.docker.com/engine/reference/builder/#entrypoint#entrypoint), otherwise you get an error (see on [stackoverflow](https://stackoverflow.com/questions/51508150/standard-init-linux-go190-exec-user-process-caused-no-such-file-or-directory)).